File: backend/User/src/main/java/tn/esprit/user/dto/UserDto.java
package tn.esprit.user.dto;

import lombok.Data;
import tn.esprit.user.entity.UserRole;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Set;

@Data
public class UserDto {
    private String userId;
    private String name;
    private String email;
    private UserRole role;
    private String phone;
    private boolean isVerified;
    private LocalDateTime createdAt;

    // Common profile fields
    private LocalDate dateOfBirth;
    private String emergencyContact;
    private String profilePicture;

    // Doctor fields
    private Integer yearsExperience;
    private String specialization;
    private String medicalLicense;
    private String workplaceType;
    private String workplaceName;

    // Relationship fields (emails of connected users)
    private Set<String> caregiverEmails;   // for patient
    private Set<String> patientEmails;
    private String doctorEmail;// for caregiver
}

File: user-node-service/src/users/dto/user.dto.ts
import { UserRole } from '../schemas/user-role.enum';
import { MedicalRecordSummaryDto } from './medical-record-summary.dto';

export class UserDto {
  userId: string;
  name: string;
  email: string;
  role: UserRole;
  phone?: string;
  isVerified: boolean;
  createdAt: Date;

  // Common profile fields
  dateOfBirth?: Date;
  emergencyContact?: string;
  profilePicture?: string;

  // Doctor fields
  yearsExperience?: number;
  specialization?: string;
  medicalLicense?: string;
  workplaceType?: string;
  workplaceName?: string;
  doctorEmail?: string;

  // Relationship fields
  caregiverEmails?: string[]; // for patient
  patientEmails?: string[]; // for caregiver

  medicalRecord?: MedicalRecordSummaryDto;
}
